extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/MSQTaskSqlEngine.java

```diff
@@ -92,7 +93,7 @@ public class MSQTaskSqlEngine implements SqlEngine
   @Inject
   public MSQTaskSqlEngine(final OverlordClient overlordClient, final ObjectMapper jsonMapper)
   {
-    this(overlordClient, jsonMapper, new MSQTerminalStageSpecFactory());
+    this(overlordClient, jsonMapper, new NewSegmentsStageSpecFactory());
```

Comment:
This factory should come from the MSQ module. I donot think a new is required here no ?

Reply:
Removed this constructor